### ISSUE TITLE
Can't update Podfile (Undefined method real_path)

### ISSUE THREAD

**Issue by patreu22** [CocoaPods/CocoaPods#6136]
Hello, I made the mistake to update to Cocoapods 1.1.1 while using XCode 8.
After recognizing my mistake I tried to go back to 1.1.0 but I can't stop pods giving me this command line output:

### Command

```
/usr/local/bin/pod update
```

### Report

* What did you do?
Tried to update my Podfile

* What did you expect to happen?
An update

* What happened instead?
The following errors

### Stack

```
   CocoaPods : 1.1.0
        Ruby : ruby 2.0.0p648 (2015-12-16 revision 53162) [universal.x86_64-darwin16]
    RubyGems : 2.6.8
        Host : Mac OS X 10.12 (16A323)
       Xcode : 8.0 (8A218a)
         Git : git version 2.8.4 (Apple Git-73)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 2f068312aaf7531183568f51b668d27f1bf397a8
```

### Plugins

```
cocoapods-deintegrate : 1.0.1
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.1.1
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
# Uncomment this line to define a global platform for your project
 platform :ios, '8.0'
# Uncomment this line if you're using Swift
 use_frameworks!

target '<ProjectName>' do
#  pod 'Armchair'
	pod 'Charts'
	pod 'Firebase/Core'
end

post_install do |installer|
    installer.pods_project.targets.each do |target|
        target.build_configurations.each do |config|
            config.build_settings['SWIFT_VERSION'] = '3.0'
        end
    end
  end
```

### Error

```
Nanaimo::Reader::ParseError - [!] Unexpected end of string while parsing
 #  -------------------------------------------
1>  
    ^
 #  -------------------------------------------

/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.2.2/lib/nanaimo/reader.rb:284:in `raise_parser_error'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.2.2/lib/nanaimo/reader.rb:129:in `parse_object'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.2.2/lib/nanaimo/reader.rb:106:in `parse!'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.4.1/lib/xcodeproj/plist.rb:27:in `read_from_path'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.4.1/lib/xcodeproj/project.rb:200:in `initialize_from_file'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.4.1/lib/xcodeproj/project.rb:102:in `open'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0/lib/cocoapods/installer/analyzer.rb:855:in `block (2 levels) in inspect_targets_to_integrate'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0/lib/cocoapods/installer/analyzer.rb:854:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0/lib/cocoapods/installer/analyzer.rb:854:in `block in inspect_targets_to_integrate'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0/lib/cocoapods/user_interface.rb:64:in `section'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0/lib/cocoapods/installer/analyzer.rb:849:in `inspect_targets_to_integrate'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0/lib/cocoapods/installer/analyzer.rb:66:in `analyze'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0/lib/cocoapods/installer.rb:236:in `analyze'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0/lib/cocoapods/installer.rb:150:in `block in resolve_dependencies'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0/lib/cocoapods/user_interface.rb:64:in `section'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0/lib/cocoapods/installer.rb:149:in `resolve_dependencies'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0/lib/cocoapods/installer.rb:110:in `install!'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0/lib/cocoapods/command/update.rb:60:in `run'
/Library/Ruby/Gems/2.0.0/gems/claide-1.0.1/lib/claide/command.rb:334:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0/lib/cocoapods/command.rb:50:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0/bin/pod:55:in `<top (required)>'
/usr/local/bin/pod:22:in `load'
/usr/local/bin/pod:22:in `<main>'
```

**Comment by patreu22** [comment]
After reading issue #6129 I updated Nanaimo.
Now I get the following error:

### Command

```
/usr/local/bin/pod update
```

### Report

* What did you do?

* What did you expect to happen?

* What happened instead?


### Stack

```
   CocoaPods : 1.1.0
        Ruby : ruby 2.0.0p648 (2015-12-16 revision 53162) [universal.x86_64-darwin16]
    RubyGems : 2.6.8
        Host : Mac OS X 10.12 (16A323)
       Xcode : 8.0 (8A218a)
         Git : git version 2.8.4 (Apple Git-73)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 2f068312aaf7531183568f51b668d27f1bf397a8
```

### Plugins

```
cocoapods-deintegrate : 1.0.1
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.1.1
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
# Uncomment this line to define a global platform for your project
 platform :ios, '8.0'
# Uncomment this line if you're using Swift
 use_frameworks!

target 'SBF Lehrer' do
#  pod 'Armchair'
	pod 'Charts'
	pod 'Firebase/Core'
end

post_install do |installer|
    installer.pods_project.targets.each do |target|
        target.build_configurations.each do |config|
            config.build_settings['SWIFT_VERSION'] = '3.0'
        end
    end
  end
```

### Error

```
NoMethodError - undefined method `real_path' for nil:NilClass
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.4.1/lib/xcodeproj/project/object/container_item_proxy.rb:96:in `container_portal_annotation'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.4.1/lib/xcodeproj/project/object/container_item_proxy.rb:105:in `to_hash_as'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.4.1/lib/xcodeproj/project/object.rb:421:in `to_ascii_plist'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.4.1/lib/xcodeproj/project.rb:287:in `block in to_ascii_plist'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.4.1/lib/xcodeproj/project.rb:285:in `each'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.4.1/lib/xcodeproj/project.rb:285:in `to_ascii_plist'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.4.1/lib/xcodeproj/project.rb:353:in `block in save'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.4.1/lib/xcodeproj/project.rb:353:in `open'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.4.1/lib/xcodeproj/project.rb:353:in `save'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0/lib/cocoapods/installer/user_project_integrator.rb:139:in `block in save_projects'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0/lib/cocoapods/installer/user_project_integrator.rb:137:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0/lib/cocoapods/installer/user_project_integrator.rb:137:in `save_projects'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0/lib/cocoapods/installer/user_project_integrator.rb:65:in `integrate!'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0/lib/cocoapods/installer.rb:612:in `block in integrate_user_project'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0/lib/cocoapods/user_interface.rb:64:in `section'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0/lib/cocoapods/installer.rb:609:in `integrate_user_project'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0/lib/cocoapods/installer.rb:117:in `install!'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0/lib/cocoapods/command/update.rb:60:in `run'
/Library/Ruby/Gems/2.0.0/gems/claide-1.0.1/lib/claide/command.rb:334:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0/lib/cocoapods/command.rb:50:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0/bin/pod:55:in `<top (required)>'
/usr/local/bin/pod:22:in `load'
/usr/local/bin/pod:22:in `<main>'
```

**Comment by patreu22** [comment]
I started to reset everything, now I'm getting again the first error throw and can't fix it by updating nanaimo.

I also found out that it targets every command I try, doesn't matter if it's `pod update`, `pod install` or `pod deintegrate`

**Comment by segiddins** [comment]
Please share the project that you're trying to integrate into so the team can investigate. Thanks!

**Comment by patreu22** [comment]
I'm sorry but due to my company's restrictions I can't share the code with you guys :/

**Comment by segiddins** [comment]
Unless you can create another project that reproduces the issue, I'm afraid there's no way for us to look into it.

**Comment by romapires** [comment]
I've come across the 

> Nanaimo::Reader::ParseError - [!] Unexpected end of string while parsing

 problem.

I've went on to do some ruby-noob debugging of Nanaimo, assuming first that the problem is not Nanaimo but something it's being fed, as it finds and EOS in first char of line 1

/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.4.1/lib/xcodeproj/plist.rb:27:in `read_from_path'
this is what is calling Nanaimo so I put on some prints there

>print "SLB"
print path
print "SLB"
Nanaimo::Reader.new(contents).parse!.as_ruby`

the file at the end of the path was empty for some reason, hence I checked out the previous version and it worked like a charm.

hope it goes the same way for you.
